### PR TITLE
BGDIINF_SB-1412: filter items by bbox

### PR DESCRIPTION
**Pull request by rebert**
Filtering items by bbox

### CODE REVIEW

File: app/stac_api/views.py [hunk 1]
```diff
@@ -152,13 +153,24 @@ def get_queryset(self):
...
+            queryset = self.filter_by_bbox(queryset, bbox)
 
         if date_time:
             queryset = self.filter_by_datetime(queryset, date_time)
 
         return queryset
 
+    def filter_by_bbox(self, queryset, bbox):
+        try:
+            logger.debug(bbox)
```
Comment (ltshb) [inline app/stac_api/views.py:165]: I would be a bit more verbose here in the log or remove it, e.g. 'filter with bbox %s', bbox.

File: app/stac_api/views.py [hunk 2]
```diff
@@ -152,13 +153,24 @@ def get_queryset(self):
...
+            query_bbox_polygon = Polygon.from_bbox(bbox.split(','))
+        except:
```
Comment (ltshb) [inline app/stac_api/views.py:167]: Never ever catch all exceptions ! Only catch known exceptions e.g. 'ValueError', see https://github.com/geoadmin/doc-guidelines/blob/master/PYTHON.md#7-error-handling---rules-of-thumb

**Review by ltshb — CHANGES_REQUESTED** [review]
Exceptions handling needs to be corrected based on https://github.com/geoadmin/doc-guidelines/blob/master/PYTHON.md#7-error-handling---rules-of-thumb

File: app/stac_api/views.py [hunk 3]
```diff
@@ -152,13 +153,24 @@ def get_queryset(self):
...
+            raise ValidationError(_('Invalid bbox query parameter, '
```
Comment (ltshb) [inline app/stac_api/views.py:168]: also all error should be logged

File: app/stac_api/views.py (hunk 2 above)
Comment (rebert) [reply app/stac_api/views.py:166]: Thanks - I will aprove

File: app/stac_api/views.py (hunk 1 above)
Comment (rebert) [reply app/stac_api/views.py]: this was a develop logger - I removed it

**Review by ltshb — CHANGES_REQUESTED** [review]
:+1:  however the openapi.yaml should be directly edited ! it is generated with `make build-spec`

**Review by ltshb — COMMENTED** [review]
Very nice :+1: , I've noticed that unittest is missing. Because the query parameter is a user input, the unittest is quite important. We need to test all use cases in unittest, not only the good one but also the border case (e.g. bbox=foo or bbox= or bbox=2).
Another point is the ValidationError, which could be improved by giving a real example: `bbox=5.96,45.82,10.49,47.81`, this it gives a hint on the syntax.

**Review by ltshb — COMMENTED** [review]
Maybe also change the example in the spec `components/schemas.yml` line 103 to reflect switzerland instead of NZ :-)

**Review by boecklic — CHANGES_REQUESTED** [review]
Nice! Same comments about unit tests and openapi*.yaml as Brice

File: app/tests/test_items_endpoint.py [hunk 4]
```diff
@@ -215,3 +216,32 @@ def test_items_endpoint_datetime_invalid_range_query(self):
...
+
+    def test_items_endpoint_bbox_valid_query(self):
+        # test bbox
+        response = self.client.get(
+            f"/{API_BASE}collections/{self.collections[0].collection_name}/items"
+            f"?bbox=5.96,45.82,10.49,47.81&limit=100"
+        )
+        json_data = response.json()
+        self.assertEqual(200, response.status_code, msg=get_description(json_data))
+        self.assertEqual(3, len(json_data['features']), msg="More than one item found")
```
Comment (ltshb) [inline app/tests/test_items_endpoint.py:229]: Message is wrong, it should be `More or less than 3 item found`

**Review by ltshb — APPROVED** [review]
:+1:  just a minor issue i unittest message

**Pull request merged**